configure.ac

```diff
@@ -52,14 +52,24 @@ AC_ARG_WITH([nccl],
 AC_ARG_WITH([mpi],
             AC_HELP_STRING([--with-mpi=PATH], [Path to non-standard MPI installation]),
             [AS_IF([test -d $withval/lib64], [mpi_libdir="lib64"], [mpi_libdir="lib"])
-             CFLAGS="-I$withval/include $CFLAGS"
+             MPIINC="$withval/include"
+             CFLAGS="-I$MPIINC $CFLAGS"
              LDFLAGS="-L$withval/$mpi_libdir $LDFLAGS"],
             [])
 
+
 AC_ARG_ENABLE([tests], [AS_HELP_STRING([--disable-trace],
       [Disable build of test binaries])], [enable_tests=$enableval], [])
 AM_CONDITIONAL([ENABLE_TESTS], [test "x$enable_tests" != "xno"])
```

Comment:
Unsure whether to keep this AM_CONDITIONAL, or to move it to the AS_IF below.

Reply:
Yes; you never want AM_CONDITIONALs to be in shell conditionals.  Usually Automake will warn if it can detect that.